pkg/k8s/volumes/crud.go

```diff
@@ -1,32 +1,58 @@
 package volumes
 
 import (
+	"context"
 	"fmt"
 	"strings"
+	"time"
 
 	"github.com/okteto/okteto/pkg/log"
 	"github.com/okteto/okteto/pkg/model"
 
+	apiv1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
 	"k8s.io/client-go/kubernetes"
 )
 
+const (
+	maxRetries = 150
+)
+
 //Create deploys the volume claim for a given dev environment
-func Create(name string, dev *model.Dev, c *kubernetes.Clientset) error {
+func Create(ctx context.Context, name string, dev *model.Dev, c *kubernetes.Clientset) error {
 	vClient := c.CoreV1().PersistentVolumeClaims(dev.Namespace)
 	pvc := translate(name)
 	k8Volume, err := vClient.Get(pvc.Name, metav1.GetOptions{})
 	if err != nil && !strings.Contains(err.Error(), "not found") {
 		return fmt.Errorf("error getting kubernetes volume claim: %s", err)
 	}
-	if k8Volume.Name != "" {
-		return nil
+	if k8Volume.Name == "" {
+		log.Infof("creating volume claim '%s'...", pvc.Name)
+		k8Volume, err = vClient.Create(pvc)
+		if err != nil {
+			return fmt.Errorf("error creating kubernetes volume claim: %s", err)
+		}
 	}
 
-	log.Infof("creating volume claim '%s'...", pvc.Name)
-	_, err = vClient.Create(pvc)
-	if err != nil {
-		return fmt.Errorf("error creating kubernetes volume claim: %s", err)
+	tries := 0
+	ticker := time.NewTicker(1 * time.Second)
+	for k8Volume.Status.Phase != apiv1.ClaimBound {
```

Comment:
would this also catch the situation where the PVC is not scheduled due to lack of storage, or lack of storage class (like what happened in k3s) and stuff like that?

Reply:
I don´t remember the k3s case well. if the claim is not bound, this will fail.